app/utils/post/index.ts

```diff
@@ -39,7 +40,8 @@ export function isPostEphemeral(post: PostModel): boolean {
 }
 
 export function isPostPendingOrFailed(post: PostModel): boolean {
-    return post.pendingPostId === post.id || post.props?.failed;
+    const isFailed = post.props?.failed || ((post.pendingPostId === post.id) && (Date.now() > post.updateAt + POST_TIME_TO_FAIL));
```

Comment:
To avoid repetition, we can make this a function (we are using this also on `post_list/post/post`)

Reply:
done